Stack/Opc.Ua.Core/Stack/Tcp/TcpTransportListener.cs

```diff
@@ -481,8 +523,15 @@ private void OnAccept(object sender, SocketAsyncEventArgs e)
                         return;
                     }
 
+                    bool serveChannel = !(m_maxChannelCount > 0 && m_maxChannelCount < m_channels.Count);
+                    if (!serveChannel)
+                    {
+                        Utils.LogError("OnAccept: Maximum number of channels {0} reached, serving channels is stopped until number is lower or equal than {1} ",
+                            m_channels.Count, m_maxChannelCount);
+                    }
+
                     // check if the accept socket has been created.
-                    if (e.AcceptSocket != null && e.SocketError == SocketError.Success)
+                    if (serveChannel && e.AcceptSocket != null && e.SocketError == SocketError.Success)
                     {
```

Comment:
shouldn't the ACceptSocket be disposed in this case?

```suggestion
                        if (!serveChannel)
                        {
                            Utils.LogError("OnAccept: Maximum number of channels {0} reached, serving channels is stopped until number is   lower or equal than {1} ",
                                m_channels.Count, m_maxChannelCount);
                            Utils.Dispose(e.AcceptSocket);    
                        }

```

Reply:
It doesn't "hurt" but the reason I did not is that `e.Dispose()` is called anyway later at line 585 and expected it to dispose the AcceptSocket implicitly.

Reply:
current fix sounds good to me